### ISSUE TITLE
Cloning a model with the "Add" layer - Custom mask layers require a config and must override get_config

### ISSUE THREAD

**Issue by rustequal** [tensorflow/tensorflow#49754]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 18.04.5 TLS (Google Colab)
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.5.0

**Description**

If the model with the "Add" layer is a part of another model, then when I try to clone it, the following warning appears:

```
CustomMaskWarning: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument.
```

This warning appeared only in version 2.5.0

**Code**

```
import tensorflow as tf
from tensorflow.keras.layers import Input, Add
from tensorflow.keras.models import Model

input1 = Input(1)
input2 = Input(1)
x = Add()([input1, input2])
model_a = Model([input1, input2], x)

input3 = Input(1)
input4 = Input(1)
x = model_a([input3, input4])
model_b = Model([input3, input4], x)

model_b_clone = tf.keras.models.clone_model(model_b)
```

**Output**

```
/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/utils/generic_utils.py:497: CustomMaskWarning: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument.
  category=CustomMaskWarning)
```

**Comment by tilakrayal** [comment]
I was able to reproduce the issue in TF v2.5 and nightly.Executed in v2.4 without error.Please find of gist [here](https://colab.research.google.com/gist/tilakrayal/36ba13908e4e968e7da4e61df2dc2569/49754.ipynb).

**Comment by Saduf2019** [comment]
@rustequal 
Please update as per above comment, we do not face any error in tf 2.4 which is stable version of tf.

**Comment by rustequal** [comment]
> @rustequal
> Please update as per above comment, we do not face any error in tf 2.4 which is stable version of tf.

yes, I confirm it. Since TF 2.5.0 is installed in Google Colab by default, I have to roll back to version 2.4.1. Hopefully this bug will be fixed in TF 2.5.

**Comment by osma** [comment]
It's not just cloning. Also saving a model with an Add layer triggers this warning. Example adapted from above:

```python
from tensorflow.keras.layers import Input, Add
from tensorflow.keras.models import Model

input1 = Input(1)
input2 = Input(1)
x = Add()([input1, input2])
model_a = Model([input1, input2], x)
model_a.save('model_a')
```

**Comment by jvishnuvardhan** [comment]
@rustequal Agree with you that it was an issue with old versions. 

This was resolved in recent `tf-nightly`.[Here](https://colab.research.google.com/gist/jvishnuvardhan/a8ca3bd6772f390b855cf765b4df7eaa/untitled.ipynb) is a gist for reference. If you want stable version, then it will be available in the upcoming `TF2.8` in near future. Thanks!

I am closing this issue as this was resolved. Please feel free to reopen if the issue persists again. Thanks!
